src/processors/physics.rs

```diff
@@ -0,0 +1,127 @@
+//! Physics processor based on `nphysics3d`
+
+extern crate ncollide;
+extern crate nphysics3d;
+
+use std::ops::{Deref, DerefMut};
+
+use components::physics::PhysicsComponent;
+use components::transform::LocalTransform;
+use world_resources::Time;
+
+use ecs::{Join, Processor, RunArg};
+use self::nphysics3d::world::World;
+
+
+/// Wrapper around `nphysics`'s `World` that impls `Send`/`Sync`, so that
+/// it can be processed by the ECS system. We take care of ensuring that
+/// there are no data races inside the ECS system, but the compiler can't
+/// infer this, so we have to manually tell it that it's ok.
+pub struct PhysicsWorld(World<f32>);
+
+impl PhysicsWorld {
+    pub fn new() -> PhysicsWorld {
+        PhysicsWorld(World::new())
+    }
+}
+
+unsafe impl Sync for PhysicsWorld { }
+unsafe impl Send for PhysicsWorld { }
+
+impl Deref for PhysicsWorld {
+    type Target = World<f32>;
+    fn deref(&self) -> &World<f32> {
+        &self.0
+    }
+}
+
+impl DerefMut for PhysicsWorld {
+    fn deref_mut(&mut self) -> &mut World<f32> {
+        &mut self.0
+    }
+}
+
+
+/// Handles physics processing for ECS
+pub struct PhysicsProcessor;
+
+impl PhysicsProcessor {
+    pub fn new() -> PhysicsProcessor {
+        PhysicsProcessor { }
+    }
+}
+
+impl Processor<()> for PhysicsProcessor {
+    fn run(&mut self, arg: RunArg, _: ()) {
+        let (
+            mut world,
+            mut handles,
+            mut locals,
+            time,
+        ) = arg.fetch(|w| (
+            w.write_resource::<PhysicsWorld>(),
+            w.write::<PhysicsComponent>(),
+            w.write::<LocalTransform>(),
+            w.read_resource::<Time>(),
+        ));
+
+        // Update physics world state
+        world.step(time.delta_time.subsec_nanos() as f32 / 1.0e9);
+
+        // Update the `LocalTransform` for each entity that has a
+        // physics component.
+        for (handle, local) in (&mut handles, &mut locals).iter() {
+            let body = handle.borrow();
+            let position = body.position();
+
+            // Update translation
+            local.translation[0] = position.translation[0];
+            local.translation[1] = position.translation[1];
+            local.translation[2] = position.translation[2];
+
+            // Update rotation. `nalgebra` stores rotation as a rotation matrix,
```

Comment:
This is pretty unfortunate to include this code. Perhaps, we could ping @sebcrozet to publish a minor version bump of `nalgebra` with this function?